swap_meet/vendor.py

```diff
@@ -0,0 +1,87 @@
+from swap_meet.item import Item
+
+
+class Vendor:
+   
+    def __init__(self, inventory = None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+    def add(self,item):
+        """ Adds item to inventory if item not 
+        in inventory """
+
+        if item not in self.inventory:
+            self.inventory.append(item)
+            return item
+    
+    def remove(self,item):
+        """Removes item from inventory if 
+        item in inventory """
+        if item in self.inventory:
+            self.inventory.remove(item)
+            
+            return item
+    
+        return False
+    
+    def get_by_category(self,category):
+        """Takes in cateogry in string form and
+        returns a list of items from inventory
+        that fall under that category """
+
+        category_list=[]
+        
+        for item in self.inventory:
+            if item.category==category:
+                category_list.append(item)
+
+        return category_list
+
+    def swap_items(self,friend,item_b,item_d):
+        """ Swaps items between Self and friend's inventory"""
+
+        if item_b in self.inventory and item_d in friend.inventory:
+            self.inventory.remove(item_b)
+            friend.inventory.append(item_b)
+            friend.inventory.remove(item_d)
+            self.inventory.append(item_d)
+        
+            return True
+     
+        return False
+
+
+    def swap_first_item(self,friend):
+        """ Swaps first item between self and friend's inventory if both inventories are not empty"""
+        if self.inventory != [] and friend.inventory != []:
+            self.swap_items(friend,self.inventory[0],friend.inventory[0])
+            
+            return True
+    
+        return False
+
+
+    def get_best_by_category(self,category):
+        """ Obtains best item from inventory that falls under select category """
+      
+        i_list=[]
+        for object in self.get_by_category(category):
+            i_list.append(object.condition)
+        for object in self.get_by_category(category):
+            if object.condition==max(i_list):
+                return object
```

Comment:
👍 this works just fine! we could merge both of these for loops into one, though:

```suggestion
    def get_best_by_category(self,category):
        """ Obtains best item from inventory that falls under select category """
      
        best_item = self.inventory[0]
        for object in self.get_by_category(category):
          if object.condition > best_item.condition:
                best_item = object
        return object
```

we set a variable to the first item in the inventory and compare each item to it.